docs/customization/api.rst

```diff
@@ -95,4 +105,362 @@ The next step is to modify the security configuration in ``config/packages/secur
+        //...
+    }
+
+.. tip::
+
+    Read more about API Platform `serialization groups <https://api-platform.com/docs/core/serialization/#using-serialization-groups>`_
+
+
+We were able to add a field that exists in ``Product`` class, but what if you want to extend it with custom fields?
+Let's customize response now with your custom fields serialized in response.
+
+Adding a custom field to response
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's say that you want to add a new field named ``additionalText`` to ``Product``.
+First we need to create a new serializer that will support our ``Product`` resource, but in this case we have a ``ProductNormalizer`` provided from Sylius.
+Unfortunately we cannot use more than one normalizer per resource, hence we will override existing one.
+
+Let's than copy code of ProductNormalizer from ``vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Serializer/ProductNormalizer.php`` :
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Serializer;
+
+    use Sylius\Component\Core\Model\ProductInterface;
+    use Symfony\Component\Serializer\Normalizer\ContextAwareNormalizerInterface;
+    use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+    use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
+    use Webmozart\Assert\Assert;
+
+    final class ProductNormalizer implements ContextAwareNormalizerInterface, NormalizerAwareInterface
+    {
+        use NormalizerAwareTrait;
+
+        private const ALREADY_CALLED = 'product_normalizer_already_called';
+
+        public function normalize($object, $format = null, array $context = [])
+        {
+            Assert::isInstanceOf($object, ProductInterface::class);
+            Assert::keyNotExists($context, self::ALREADY_CALLED);
+
+            $context[self::ALREADY_CALLED] = true;
+
+            $data = $this->normalizer->normalize($object, $format, $context);
+            $variant = $this->defaultProductVariantResolver->getVariant($object);
+            $data['defaultVariant'] = $variant === null ? null : $this->iriConverter->getIriFromItem($variant);
+
+            return $data;
+        }
+
+        public function supportsNormalization($data, $format = null, $context = []): bool
+        {
+            if (isset($context[self::ALREADY_CALLED])) {
+                return false;
+            }
+
+            return $data instanceof ProductInterface;
+        }
+    }
+
+And now let's declare its service in config files:
+
+.. code-block:: yaml
+
+    # config/services.yaml
+    App\Serializer\ProductNormalizer:
+        tags:
+            - { name: 'serializer.normalizer', priority: 100 }
+
+.. warning::
+
+    As we can use only one Normalizer per resource we need to set priority higher then one from Sylius.
+    Default value for Sylius Normalizers is typically 64, but if you want to be sure, check the values in ``src/Sylius/Bundle/ApiBundle/Resources/config/services/serializers.xml``
```

Comment:
I'm not sure if 64 is a typical value, we've probably just copied it from API Platform docs